client/src/templates/Challenges/classic/editor.tsx

```diff
@@ -409,7 +409,65 @@ const Editor = (props: EditorProps): JSX.Element => {
         });
       }
     });
+    // Introduced as a work around for a bug in JAWS 2022
+    // https://github.com/FreedomScientific/VFO-standards-support/issues/598
+    editor.addAction({
+      id: 'toggle-aria-roledescription',
+      label: 'Toggle aria-roledescription',
+      keybindings: [
+        monaco.KeyMod.CtrlCmd | monaco.KeyMod.Alt | monaco.KeyCode.KEY_J
+      ],
+      run: toggleAriaRoledescription
+    });
     editor.onDidFocusEditorWidget(() => props.setEditorFocusability(true));
+
+    // aria-roledescription is on (true) by default, check if it needs
+    // to be removed.
+    const ariaRoledescription = getStoredAriaRoledescription();
+    if (!ariaRoledescription) {
+      setAriaRoledescription(false);
+    }
+  };
+
+  const toggleAriaRoledescription = () => {
+    const newRoledescription = !getStoredAriaRoledescription();
+    setAriaRoledescription(newRoledescription);
+    ariaAlert(
+      `aria-roledescription has been turned ${
+        newRoledescription ? 'on' : 'off'
+      }`
+    );
+  };
+
+  const setAriaRoledescription = (value: boolean) => {
+    const textareas = document.querySelectorAll('.monaco-editor textarea');
+    textareas.forEach(textarea => {
+      value
+        ? textarea.setAttribute('aria-roledescription', 'editor')
+        : textarea.removeAttribute('aria-roledescription');
+    });
+    store.set('ariaRoledescription', value);
+  };
+
+  const getStoredAriaRoledescription = (): boolean => {
+    const roledescription = store.get('ariaRoledescription') as
+      | boolean
+      | undefined;
+    return !!(roledescription === undefined || roledescription);
+  };
+
+  // Borrowed from
+  // freeCodeCamp/node_modules/monaco-editor/esm/vs/base/browser/ui/aria/aria.js
+  const ariaAlert = (message: string) => {
+    const ariaLive = document.getElementsByClassName(
+      'monaco-alert'
+    ) as HTMLCollection;
+    if (ariaLive.length > 0) {
+      const liveText = ariaLive[0] as HTMLElement;
+      liveText.textContent = message;
+      liveText.style.visibility = 'hidden';
+      liveText.style.visibility = 'visible';
```

Comment:
```suggestion
      liveText.style.visibility = 'visible';
```
Doesn't the second assignment just wipe out the first?

Reply:
Yes it does, but this is a hack being used by the monaco editor and thus we probably need to use it as well if we are going to use the aria live region provided by monaco. You can read more about the hack at [HTML5 Accessibility Chops: ARIA role=alert browser support](https://www.tpgi.com/html5-accessibility-chops-aria-rolealert-browser-support/). You can see it in action in the monaco file I referenced in the comment above the function declaration (it's in the `insertMessage` function). FYI, this hack is extremely old (the date on the TPGI article is 2012) and is probably not needed today, but since that's the way it is still implemented in monaco, I think we should do that as well here.

I did initially think about just importing the `alert` function from that monaco file. I don't think it was intended to be a public API but I did test it and it seemed to work. If you would rather we do that I will make the change. But then TS is going to complain about not finding a declaration file for the module.

I looked to see if FCC had its own aria live region but I didn't find any. If it does, please let me know and I will use that one instead.

Reply:
This is all totally fine. It's a bit hacky, as you say, but that's fine (in part because this is intended to be temporary).

If you could put a brief comment explaining why this is done (to make the element suddenly 'appear' and force screenreaders to read it, right?) that should stop future devs from thinking this can be cleaned up.